scripts/gen_operator_csv.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+#
+# Generate an operator bundle for publishing to OLM. Copies appropriate files
+# into a directory, and composes the ClusterServiceVersion which needs bits and
+# pieces of our rbac and deployment files.
+#
+# Usage ./scripts/gen_operator_csv.py OUTPUT_DIR PREVIOUS_VERSION IMAGE_NAME
+# Example: scripts/gen_operator_csv.py tmp 0.1 quay.io/redhat/dedicated-admin-operator:latest
+
+import datetime
+import os
+import sys
+import yaml
+import shutil
+import subprocess
+
+
+def get_git_sha():
+    sha = subprocess.check_output('git rev-parse HEAD', shell=True)
+    return str(sha)[0:7]
+
+def get_num_commits():
+    num = subprocess.check_output('git rev-list HEAD --count', shell=True)
+    return num.rstrip()
+
+
+if __name__ == '__main__':
+
+    # This script will append the current number of commits given as an arg
+    # (presumably since some past base tag), and the git hash arg for a final
+    # version like: 0.1.189-3f73a592
+    VERSION_BASE = "0.1"
+
+    if len(sys.argv) != 4:
+        print("USAGE: %s OUTPUT_DIR PREVIOUS_VERSION IMAGE_NAME" % sys.argv[0])
+        sys.exit(1)
+
+    outdir = sys.argv[1]
+    prev_version = sys.argv[2]
```

Comment:
This needs to be optional because building a `CatalogSource` image fails if the specified `prev_version` isn't (also) in the catalog:

```
# image build
validate manifests/dedicated-admin-operator/0.1.28-8bbbc8f/dedicated-admin-operator.v0.1.28-8bbbc8f.clusterserviceversion.yaml
validate manifests/dedicated-admin-operator/dedicated-admin-operator.package.yaml
time="2019-03-25T16:40:35Z" level=info msg="validating manifests" dir=manifests
time="2019-03-25T16:40:35Z" level=info msg="loading Bundles" dir=manifests
time="2019-03-25T16:40:35Z" level=info msg=directory dir=manifests file=manifests load=bundles
time="2019-03-25T16:40:35Z" level=info msg=directory dir=manifests file=dedicated-admin-operator load=bundles
time="2019-03-25T16:40:35Z" level=info msg=directory dir=manifests file=0.1.28-8bbbc8f load=bundles
time="2019-03-25T16:40:35Z" level=info msg="found csv, loading bundle" dir=manifests file=dedicated-admin-operator.v0.1.28-8bbbc8f.clusterserviceversion.yaml load=bundles
time="2019-03-25T16:40:35Z" level=info msg="loading bundle file" dir=manifests file=dedicated-admin-operator.v0.1.28-8bbbc8f.clusterserviceversion.yaml load=bundle
time="2019-03-25T16:40:35Z" level=info msg="skipping non-csv file" dir=manifests file=dedicated-admin-operator.package.yaml load=bundles
time="2019-03-25T16:40:35Z" level=info msg="loading Packages" dir=manifests
time="2019-03-25T16:40:35Z" level=info msg=directory dir=manifests file=manifests load=package
time="2019-03-25T16:40:35Z" level=info msg=directory dir=manifests file=dedicated-admin-operator load=package
time="2019-03-25T16:40:35Z" level=info msg=directory dir=manifests file=0.1.28-8bbbc8f load=package
time="2019-03-25T16:40:35Z" level=info msg="skipping non-package file" dir=manifests file=dedicated-admin-operator.v0.1.28-8bbbc8f.clusterserviceversion.yaml load=package
time="2019-03-25T16:40:35Z" level=fatal msg="error loading package into db:  specifies replacement that couldn't be found"
```

Reply:
according to slack discussing we understand that leaving the prev_version field as optional could encourage people to leave it blank